internal/bminventory/inventory.go

```diff
@@ -1668,6 +1676,107 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
+	// case 5
+	if platform == nil && userManagedNetworking == nil {
+		return nil, nil, nil
+	}
+
+	// case 1
+	if err := checkPlatformWrongParamsInput(platform, userManagedNetworking); err != nil {
+		return nil, nil, err
+	}
+
+	if *cluster.Platform.Type == models.PlatformTypeBaremetal {
+
+		// case 2
+		if !swag.BoolValue(userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeBaremetal) {
+			return nil, nil, nil
+		}
+
+		// case 3
+		if (platform != nil && *platform.Type == models.PlatformTypeNone) || (swag.BoolValue(userManagedNetworking) && platform == nil) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
+	}
+
+	if *cluster.Platform.Type == models.PlatformTypeNone {
+
+		// case 4
+		if (userManagedNetworking == nil || *userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeNone) {
+			return nil, nil, nil
+		}
+
+		// case 6
+		if !swag.BoolValue(userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeBaremetal) {
+			if cluster.CPUArchitecture != common.X86CPUArchitecture && !featuresupport.IsFeatureSupported(cluster.OpenshiftVersion, models.FeatureSupportLevelFeaturesItems0FeatureIDARM64ARCHITECTUREWITHCLUSTERMANAGEDNETWORKING) {
+				return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed for clusters with non-x86_64 CPU architecture"))
+			}
+
+			if *cluster.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+				return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed in single node mode"))
+			}
+
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+		}
+	}
+
+	return platform, userManagedNetworking, nil
+}
+
+func getActualCreateClusterPlatformParams(platform *models.Platform, userManagedNetworking *bool, highAvailabilityMode *string) (*models.Platform, *bool, error) {
+
+	if err := checkPlatformWrongParamsInput(platform, userManagedNetworking); err != nil {
+		return nil, nil, err
+	}
+
+	if *highAvailabilityMode == models.ClusterHighAvailabilityModeFull {
+		if (platform == nil || *platform.Type == models.PlatformTypeBaremetal) && !swag.BoolValue(userManagedNetworking) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+		}
+
+		if swag.BoolValue(userManagedNetworking) || (platform != nil && *platform.Type == models.PlatformTypeNone) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
+	}
+
+	if *highAvailabilityMode == models.ClusterHighAvailabilityModeNone {
```

Comment:
Why two separate "ifs" instead of an else?

Reply:
I think it's more clearer to read it this way.
We already have a lot of conditions there and to my opinion it's more clearer to "see" the condition and not need to scroll up to see the if statement.

Reply:
I disagree because this way it's disconnected from the previous "if" even though they are mutually exclusive.  Now I need to scroll up to understand that the code can't actually reach both "ifs".

Reply:
@avishayt I think that `else if` will solve that issue and still will be more clear, WDYT?

Reply:
OK